module/data/item/templates/associatedItemData.js

```diff
@@ -0,0 +1,15 @@
+const fields = foundry.data.fields;
+
+export function associatedItem(fieldName) {
+    const field = {}
+    field[fieldName] = new fields.StringField({initial: ''})
+
+    return field
```

Comment:
I dont think a generic method in the data template is good. I would like for you to make it explicit that only associatedDiagrams are added